packages/contracts/contracts/optimistic-ethereum/libraries/bridge/OVM_EOACodeHashSet.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+
+/* External Imports */
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/EnumerableSet.sol";
+
+/**
+ * @title OVM_EOACodeHashSet
+ * @dev Helper contract used to keep track of OVM EOA contract set (OVM specific)
+ *
+ * The OVM implements a basic form of account abstraction. In effect, this means
+ * that the only type of account is a smart contract (no EOAs), and all user wallets
+ * are in fact smart contract wallets. So to check for EOA, we need to actually check if
+ * the sender is an OVM_ProxyEOA contract, which gets deployed by the ovmCREATEEOA opcode.
+ *
+ * As the OVM_ProxyEOA.sol contract source could potentially change in the future (i.e., due to a fork),
+ * here we actually track a set of possible EOA proxy contracts.
+ *
+ * Compiler used: optimistic-solc
+ * Runtime target: OVM
+ */
+contract OVM_EOACodeHashSet is Ownable {
+    // Add the EnumerableSet library
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    // Declare a Bytes32Set of code hashes
+    EnumerableSet.Bytes32Set private s_codeHasheSet;
```

Comment:
Great to see this included! 🙌 

There was/is a typo here. It should actually be named `s_codeHashesSet` or `s_codeHashSet`.